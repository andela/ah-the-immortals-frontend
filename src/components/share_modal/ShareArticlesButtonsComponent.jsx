import React, {Component} from 'react';
import PropTypes from 'prop-types';
import FacebookButtonComponent from './SVG_buttons/FacebookButtonComponent';
import TwitterButtonComponent from './SVG_buttons/TwitterButtonComponent';
import MailButtonComponent from './SVG_buttons/MailButtonComponent';


class ShareArticlesButtonsComponent extends Component {
  static propTypes = {
    article:PropTypes.object.isRequired
  };
  render() {
    const { article } = this.props;
    return (
      <div className="main-svg">
        <FacebookButtonComponent article={article} />
        <TwitterButtonComponent article={article} />
        <MailButtonComponent article={article} />
      </div>
    );
  }
}

export default ShareArticlesButtonsComponent;
