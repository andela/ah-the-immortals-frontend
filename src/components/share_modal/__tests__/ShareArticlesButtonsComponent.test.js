import { mount } from 'enzyme';
import React from 'react';
import ShareArticlesButtonsComponent from '../ShareArticlesButtonsComponent';

describe('Testing the file with all the buttons', () => {
  const props = {
    article: {}
  };
  const wrapper = mount(
    <ShareArticlesButtonsComponent {...props} />
  );
  it('Test successful mount of the component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
