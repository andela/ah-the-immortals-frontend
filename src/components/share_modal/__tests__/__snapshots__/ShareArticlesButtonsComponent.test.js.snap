// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the file with all the buttons Test successful mount of the component 1`] = `ReactWrapper {}`;
